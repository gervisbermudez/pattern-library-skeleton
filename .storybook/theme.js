import {create} from '@storybook/theming/create'

const theme = create({
  base: 'light',
  brandTitle: '#zapatillasFromMarsğŸ‘ŸğŸš€',
  brandUrl: 'https://github.com/klaufel/pattern-library-skeleton'
})

export default theme
