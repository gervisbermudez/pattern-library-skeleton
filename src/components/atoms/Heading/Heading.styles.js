import styled, {css} from 'styled-components'

export const Wrapper = styled.h1`
  font-size: ${props => props.theme.typography.heading.normal.fontSize};
  line-height: ${props => props.theme.typography.heading.normal.lineHeight};
  font-weight: ${props => props.theme.typography.heading.normal.fontWeight};
  margin: 0 0 ${props => props.theme.spacing.md};

  ${props =>
    props.size === 'big' &&
    css`
      font-size: ${props => props.theme.typography.heading.big.fontSize};
      line-height: ${props => props.theme.typography.heading.big.lineHeight};
      font-weight: ${props => props.theme.typography.heading.big.fontWeight};
    `}

  ${props =>
    props.size === 'small' &&
    css`
      font-size: ${props => props.theme.typography.heading.small.fontSize};
      line-height: ${props => props.theme.typography.heading.small.lineHeight};
      font-weight: ${props => props.theme.typography.heading.small.fontWeight};
    `}
`
