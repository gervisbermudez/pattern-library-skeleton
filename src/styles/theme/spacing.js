import tokens from '../../../tokens/json/spacing'

const spacing = {
  xxs: tokens.spacing.xxs.value,
  xs: tokens.spacing.xs.value,
  sm: tokens.spacing.sm.value,
  md: tokens.spacing.md.value,
  lg: tokens.spacing.lg.value,
  xl: tokens.spacing.xl.value,
  xxl: tokens.spacing.xxl.value
}

export default spacing
