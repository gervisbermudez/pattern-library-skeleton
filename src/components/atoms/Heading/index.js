import React from 'react'
import PropTypes from 'prop-types'
import {Wrapper} from './Heading.styles'

export const HEADINGS_TYPES = {
  h1: 'h1',
  h2: 'h2',
  h3: 'h3',
  h4: 'h4',
  h5: 'h5',
  h6: 'h6',
  span: 'span'
}

export const HEADINGS_SIZES = {
  small: 'small',
  normal: 'normal',
  big: 'big'
}

const Heading = ({
  type = HEADINGS_TYPES.h1,
  size = HEADINGS_SIZES.normal,
  children,
  ...props
}) => (
  <Wrapper as={type} size={size} {...props}>
    {children}
  </Wrapper>
)

Heading.displayName = 'Heading'

Heading.propTypes = {
  /** HTML tag to render */
  type: PropTypes.oneOf(Object.keys(HEADINGS_TYPES)),
  /** Size of heading */
  size: PropTypes.oneOf(Object.keys(HEADINGS_SIZES)),
  /** Content to be included in the heading */
  children: PropTypes.node.isRequired
}

export default Heading
