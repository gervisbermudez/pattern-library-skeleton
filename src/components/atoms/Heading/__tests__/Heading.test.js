import React from 'react'
import 'jest-styled-components'
import {render} from '@testing-library/react'
import {ThemeProvider} from 'styled-components'
import theme from '../../../../styles/theme'
import Heading, {HEADINGS_TYPES, HEADINGS_SIZES} from '..'

const HeadingTypes = () =>
  Object.keys(HEADINGS_TYPES).map(type => (
    <Heading key={type} type={type}>
      Minions ipsum chasy baboiii bappleees
    </Heading>
  ))

const HeadingSizes = () =>
  Object.keys(HEADINGS_SIZES).map(size => (
    <Heading key={size} size={size}>
      Minions ipsum chasy baboiii bappleees
    </Heading>
  ))

describe('Heading', () => {
  test('it works default', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <Heading>Minions ipsum chasy baboiii bappleees</Heading>
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with all sizes variations', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <HeadingSizes />
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })

  test('it works with all types variations', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <HeadingTypes />
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })
})
