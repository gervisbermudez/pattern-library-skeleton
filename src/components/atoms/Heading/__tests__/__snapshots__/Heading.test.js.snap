// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading it works default 1`] = `
.c0 {
  font-size: 24px;
  line-height: 28px;
  font-weight: 700;
  margin: 0 0 24px;
}

<h1
  class="c0"
>
  Minions ipsum chasy baboiii bappleees
</h1>
`;

exports[`Heading it works with all sizes variations 1`] = `
.c0 {
  font-size: 24px;
  line-height: 28px;
  font-weight: 700;
  margin: 0 0 24px;
  font-size: 18px;
  line-height: 22px;
  font-weight: 700;
}

.c1 {
  font-size: 24px;
  line-height: 28px;
  font-weight: 700;
  margin: 0 0 24px;
}

.c2 {
  font-size: 24px;
  line-height: 28px;
  font-weight: 700;
  margin: 0 0 24px;
  font-size: 28px;
  line-height: 32px;
  font-weight: 700;
}

<div>
  <h1
    class="c0"
  >
    Minions ipsum chasy baboiii bappleees
  </h1>
  <h1
    class="c1"
  >
    Minions ipsum chasy baboiii bappleees
  </h1>
  <h1
    class="c2"
  >
    Minions ipsum chasy baboiii bappleees
  </h1>
</div>
`;

exports[`Heading it works with all types variations 1`] = `
.c0 {
  font-size: 24px;
  line-height: 28px;
  font-weight: 700;
  margin: 0 0 24px;
}

<div>
  <h1
    class="c0"
  >
    Minions ipsum chasy baboiii bappleees
  </h1>
  <h2
    class="c0"
  >
    Minions ipsum chasy baboiii bappleees
  </h2>
  <h3
    class="c0"
  >
    Minions ipsum chasy baboiii bappleees
  </h3>
  <h4
    class="c0"
  >
    Minions ipsum chasy baboiii bappleees
  </h4>
  <h5
    class="c0"
  >
    Minions ipsum chasy baboiii bappleees
  </h5>
  <h6
    class="c0"
  >
    Minions ipsum chasy baboiii bappleees
  </h6>
  <span
    class="c0"
  >
    Minions ipsum chasy baboiii bappleees
  </span>
</div>
`;
